draft-ietf-dnsop-svcb-https.md

```diff
@@ -1213,8 +1214,8 @@ Because HTTPS RRs are received over an often insecure channel (DNS),
 clients MUST NOT place any more trust in this signal than if they
 had received a 307 redirect over cleartext HTTP.
 If this redirection would result in a loss of functionality (e.g. important
-resources that are only available on the "http" origin), the operator MUST
-NOT publish an HTTPS RR.
+resources or behaviors that are only available when using the "http" origin),
+the operator MUST NOT publish an HTTPS RR.
```

Comment:
If I held the pen I would have put more detail in here, but I don't have a particularly strong argument that more detail is clearly required.

Reply:
I would like to see this being more explicit.  If an action that was initiated for an HTTP resource is performed on an HTTPS resource instead, this might have unexpected consequences if the two URLs are not aliases of the same logical resource.

Reply:
IIRC, we've circled and redrawn this line several times.  The breaking point for saying that the resources must be identical is that that HTTP origin might return redirects to the HTTPS origin anyway -- in which case you're *not* finding the same things over HTTPS as over HTTP, but you're winding up at the same resource.  Hence describing it in terms of loss of functionality, rather than identity of resources or responses.

Reply:
The upgrade works by simulating a redirect, so that also lines up nicely: publishing such a record, just like setting HSTS, means you're serving a redirect at the http origin. If you wanted to do that, do that. If you didn't want to do that, don't.

Reply:
Seems like the "need" for this language keeps coming up because we're afraid a domain operator will do something stupid and add the record when a redirect will break their website.  Do we really think anybody will add the record when the redirect is not desired (or at least anybody that has actually read through enough of the RFC to see the redirect happens, which should be roughly a superset of anybody reading whatever statement we put here)? If not, any additional details will just add confusion, so I'd be happy if we just left the current language or simplified it.

Reply:
And by leave the "current language", I'm fine with the text in this change or before.  I'm just against adding more on top of that to try to add any more details on when a redirect would break a domain.

Reply:
@MikeBishop, my objection was to the extreme economy of the phrase "resources or behaviors".  I was just looking for this being a little clearer by making the "behaviors" bit a little more explicit.  Something like,

> Where the HTTP resource does not redirect to the HTTPS resource or reference the same underlying resource, as is commonplace, requests directed to the HTTP resource will be processed by the HTTPS resource with potentially unexpected results.

Reply:
> Seems like the "need" for this language keeps coming up because we're afraid a domain operator will do something stupid and add the record when a redirect will break their website

I would rather characterize my primary concern as that the description in the document should accurately reflect its technical contents.  We are deviating from what the HTTP protocol specifies, and should clearly document the nature of the deviation.  (Ideally, document the motivation for the deviation as well.)

> which should be roughly a superset of anybody reading whatever statement we put here

Right, which is why I want to make it more visible in the Introduction that there are scenarios where the mechanism is not applicable (see #374 for thoughts on how)

> my objection was to the extreme economy of the phrase "resources or behaviors"

I think that also describes my concerns.

Reply:
@martinthomson, I like that text.

Reply:
OK, I've incorporated Martin's text.